NumLib/Fem/Deformation/LinearBMatrix.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_FEM_DEFORMATION_LINEARBMATRIX_H_
+#define NUMLIB_FEM_DEFORMATION_LINEARBMATRIX_H_
+
+#include <cmath>
+
+namespace NumLib
+{
+namespace LinearBMatrix
+{
+/// Fills a B-matrix based on given shape function dN/dx values.
+template <int DisplacementDim, int NPOINTS, typename DNDX_Type,
+          typename BMatrixType>
+void computeBMatrix(DNDX_Type const& dNdx, BMatrixType& b_matrix)
+{
+    static_assert(0 < DisplacementDim && DisplacementDim <= 3,
+                  "LinearBMatrix::computeBMatrix: DisplacementDim must be in "
+                  "range [1,3].");
+
+    b_matrix.setZero();
+
+    switch (DisplacementDim)
+    {
+        case 3:
+        {
+            for (int i = 0; i < NPOINTS; ++i)
+            {
+                b_matrix(2, 2 * NPOINTS + i) = dNdx(2, i);
```

Comment:
There is a paper describing all the details of the BMatrices and Kelvin mapping;
@nagelt Would you please provide a citation?

The `std::sqrt(2)` (or similar simple expressions) are usually computed at the compile-time;
Using `static const double sqrt2 = std::sqrt(2)` in this particular function made the binary code larger by 22656 bytes. I didn't (yet) tested its performance. I shall keep it as it is now.